config/default/kustomization.yaml

```diff
@@ -16,6 +16,8 @@ bases:
 - ../crd
 - ../rbac
 - ../manager
+# [Prometheus scraping] To disable prometheus scraping configuration via a ServiceMonitor (which assumes you are using the prometheus operator) comment out the following line. To change the prometheus identifier add a kustomize path for ../prometheus/servicemonitor.yaml 
+- ../prometheus
```

Comment:
Maybe the naming is a little off here, this isn't actually installing prometheus, but the ServiceMonitor to wire up prometheus to the operator.

Since this file is used to comment/uncomment sets of features (e.g. based on whether you want the auth proxy or not), having the prometheus integration here as something you can easily turn on/off seemed like a reasonabl option. Also, by having it here it integrates with the rest of the kustomize rules (such as setting the namespace, prefixing names, ...)

Would you prefer that the prometheus integration were disabled by default? (i.e. commented out here)

Reply:
Thanks for explaining the reason behind your decision, I still think we shouldn't have Prometheus ServiceMonitor configuration here. I understand the default parameters (namespace, namePrefix,..)  will be double up but our end product is data bricks operator which users will download setup.yaml from release selection. We don't know if our users are planning to use Prometheus Operator or not and we don't want to impose them to use it. It's great to have it as additional/optional setup but not part of operator setup

Reply:
The namespace and name-prefix are a couple of the aspects that are integrated in the ServiceMonitor, but there’s also a fair bit of work to ensure that it works with the RBAC options that users could switch in/out via comments in kustomize. I take your point around not imposing the choice of Prometheus Operator on users and I completely agree. Are you happy for us to achieve this by defaulting the prometheus operator files to commented out (i.e. same default behaviour as already exists) but leaving them integrated with the existing kustomize assets to make it easy for users who want to use Prometheus Operator with the databricks operator to add this scenario (and to have it work regardless of whether they also want to customise the other options such as removing the aut proxy sidecar for metrics)? Hopefully this meets your (our) desire to not impose Prometheus Operator but still allows for it in a relatively simple way.